### DISCUSSION TITLE
how to handle error from detechGhe in GitHub's initPlatform

### DISCUSSION THREAD

**Discussion by lyonlai** [renovatebot/renovate#24660] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:36.107.1-slim

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

be able to access our protected GHE server without problem. 

### What you tried so far.

accessing our protected GHE server requires mTLS auth. This is done via hostRules. However, due to the time when the [detectGhe](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/github/index.ts#L149) function was called, the hostRules hasn't picked up just yet. Hence returning errors. 

I've done some reading inside the init logic. Wondering what might be the better solution.

* move the [setGlobalHostRules](https://github.com/renovatebot/renovate/blob/main/lib/workers/global/initialize.ts#L77) above the [initPlatform](https://github.com/renovatebot/renovate/blob/main/lib/workers/global/initialize.ts#L72C26-L72C26), and supply the mtls params via environment variable host rules ( this also needs some changes to the detectHostRulesFromEnv logic).

* or an easier solution might just be like the bitbucket platform init logic, just wrap it in a try catch like [this ](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/bitbucket/index.ts#L91-L105) and logs the error instead. After passing this point I think it should be able to pickup the needed credentials from the hostRules. But I'm not entirely sure how much the GHE version is used and whether I'll miss some feature if I didn't get this information. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
I think that setting hostRules earlier sounds cleaner

**Reply by lyonlai** [reply]
SGTM, will try it out soon

**Reply by lyonlai** [reply]
I've made it working by setting hostRules earlier. 

There are couple of changes inside that might worth separate PRs for clarity. I'll get those up soon.
